packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx

```diff
@@ -42,11 +42,24 @@ const useUtilityClasses = (classes: Partial<MultiSectionDigitalClockClasses> | u
 const MultiSectionDigitalClockRoot = styled(PickerViewRoot, {
   name: 'MuiMultiSectionDigitalClock',
   slot: 'Root',
-})<{ ownerState: PickerOwnerState }>(({ theme }) => ({
-  flexDirection: 'row',
-  width: '100%',
-  borderBottom: `1px solid ${(theme.vars || theme).palette.divider}`,
-}));
+  shouldForwardProp: (prop) => prop !== 'numberOfColumns' && prop !== 'ownerState',
+})<{ ownerState: PickerOwnerState; numberOfColumns: number }>(({ theme, numberOfColumns }) => {
+  const containerMinWidth = 56 * numberOfColumns + 1; // Increased threshold for testing
+  return {
+    flexDirection: 'row',
+    width: '100%',
+    minWidth: 56 * numberOfColumns,
+    borderBottom: `1px solid ${(theme.vars || theme).palette.divider}`,
+    containerType: 'inline-size',
+    display: 'grid',
+    gridTemplateColumns: `repeat(${numberOfColumns}, minmax(56px, 1fr))`,
+    [`@container (min-width: ${containerMinWidth}px)`]: {
+      '& > .MuiMultiSectionDigitalClockSection-root': {
+        scrollbarGutter: 'stable',
+      },
+    },
```

Comment:
Slightly weird, we can discuss with the core team if we could start using container queries. This one is to avoid shrinking the items within the columns when the scrollbar is displayed

https://github.com/user-attachments/assets/72f3568c-158f-444f-88af-5b3cd9327e86